app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,24 @@
+class SessionsController < ApplicationController
+  def new
+  end
+
+  def create
```

Comment:
Consider this code:

```ruby
def create
    session = Session.new(params[:session])

    if session.valid?
      redirect_to(plays_path)
    else
      flash.now[:danger] = session.error_summary
      render 'new'
    end
  end
```

Could `SessionsController#create` be refactored as such? I think there are some solid pros to the approach, since the controller should--to the extent this is possible--not contain much intelligence. At the most, it should route requests to certain resources/models/classes.

What if there were a _model_ called `Session` where the actual implementation details lived? Can you explore?

Reply:
FWIW, I think you have the start of that model in your sessions_helper. I tend to use helpers less, because they create methods in the global namespace. Usually if I find that I have a helper, I probably could make at least one new model out of it. Thus, I end up with more pure/plain Ruby code, which is usually easier to test as well.

Dig around a bit and see what you can come up with. We can workshop it together too, if you like.